packages/cli/test/integration/generators/datasource.integration.js

```diff
@@ -0,0 +1,179 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/cli
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+
+// Imports
+const path = require('path');
+const assert = require('yeoman-assert');
+const testlab = require('@loopback/testlab');
+
+const expect = testlab.expect;
+const TestSandbox = testlab.TestSandbox;
+
+const debug = require('../../../lib/debug')('lb4:datasource:test');
+const DataSourceGenerator = require('../../../generators/datasource');
+const generator = path.join(__dirname, '../../../generators/datasource');
+const tests = require('../lib/artifact-generator')(generator);
+const baseTests = require('../lib/base-generator')(generator);
+const testUtils = require('../../test-utils');
+
+// Test Sandbox
+const SANDBOX_PATH = path.resolve(__dirname, '..', '.sandbox');
+const sandbox = new TestSandbox(SANDBOX_PATH);
+
+// CLI Inputs
+const basicCLIInput = {
+  name: 'ds',
+};
+
+const cloudantCLIInput = {
+  name: 'ds',
+  connector: 'cloudant',
+  url: 'http://user:pass@host.com',
+  username: 'user',
+  password: 'pass',
+};
+
+const numberCLIInput = {
+  name: 'ds',
+  connector: 'db2',
+  port: '100',
+};
+
+const complexCLIInput = {
+  name: 'ds',
+  connector: 'rest',
+  options: '{"test": "value"}',
+  operations: '["get", "post"]',
+};
+
+const expectedComplexJSONOutput = {
+  name: 'ds',
+  connector: 'rest',
+  options: {test: 'value'},
+  operations: ['get', 'post'],
+};
+
+// Expected File Name
+const expectedTSFile = path.join(
+  SANDBOX_PATH,
+  'src/datasources/ds.datasource.ts',
+);
+
+const expectedJSONFile = path.join(
+  SANDBOX_PATH,
+  'src/datasources/ds.datasource.json',
+);
+
+const expectedIndexFile = path.join(SANDBOX_PATH, 'src/datasources/index.ts');
+
+// Base Tests
+describe('datasource-generator extending BaseGenerator', baseTests);
+describe('generator-loopback4:datasource', tests);
+
+describe('lb4 datasource', () => {
+  beforeEach('reset sandbox', () => sandbox.reset());
+
+  it('does not run without package.json', () => {
```

Comment:
We might need to look into separating out the common tests across artifact generators like this one. Outside of scope in context of this PR though